### ISSUE TITLE
symbol_in_winbar not work well

### ISSUE THREAD

**Issue by changyiyao** [glepnir/lspsaga.nvim#371]
### Description
`symbol_in_winbar` prints 'All servers of this buffer does not support textDocument/documentSymbol' when opening or switch between buffers whose server seems not support the LSP method.

**Expected Behavior**
Auto disable ``symbol_in_winbar` or the prints when server not support. 

**Actual Behavior**
<img width="470" alt="截屏2022-07-22 15 02 09" src="https://user-images.githubusercontent.com/766842/180382718-bf8559fa-1b0f-4672-a4ea-e227a42cde2b.png">

### Details

```
require('lspsaga').init_lsp_saga({
  border_style = "rounded",
  saga_winblend = 5,
  move_in_saga = { prev = '<c-p>', next = '<c-n>' },
  diagnostic_header = { " ", " ", " ", "ﴞ " },
  show_diagnostic_source = true,
  code_action_icon = "💡",
  code_action_lightbulb = {
      enable = true,
      sign = true,
      sign_priority = 20,
      virtual_text = false,
  },
  max_preview_lines = 20,
  finder_action_keys = {
    open = "o",
    vsplit = "x",
    split = "s",
    tabe = "t",
    quit = "q",
    scroll_down = "<C-f>",
    scroll_up = "<C-b>", -- quit can be a table
  },
  code_action_keys = {
    quit = "q",
    exec = "<CR>",
  },
  rename_action_quit = "<C-c>",
  symbol_in_winbar = {
    in_custom = true,
  --   enable = false,
  --   separator = ' ',
  --   show_file = true,
  --   click_support = false,
  },
  -- show outline
  show_outline = {
    win_position = 'right',
    -- set the special filetype in there which in left like nvimtree neotree defx
    left_with = '',
    win_width = 30,
    auto_enter = true,
    auto_preview = true,
    virt_text = '┃',
    jump_key = 'o',
    -- auto refresh when change buffer
    auto_refresh = true,
  },
  server_filetype_map = { dartls = { "dart" } },
})

```
</details>

```
NVIM v0.8.0-dev-nightly-9-g1ef84547a
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
Compilation: /Applications/Xcode_13.2.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DNVIM_TS_HAS_SET_MATCH_LIMIT -DNVIM_TS_HAS_SET_ALLOCATOR -O2 -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Ws
hadow -Wconversion -Wdouble-promotion -Wmissing-noreturn -Wmissing-format-attribute -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=always -DINCLUDE_GENERATED_DECLARATIONS -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG
_LEVEL=3 -I/Users/runner/work/neovim/neovim/build/cmake.config -I/Users/runner/work/neovim/neovim/src -I/Users/runner/work/neovim/neovim/.deps/usr/include -I/Applications/Xcode_13.2.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.1.sdk/usr/include -I/Library/Framewo
rks/Mono.framework/Headers -I/Users/runner/work/neovim/neovim/build/src/nvim/auto -I/Users/runner/work/neovim/neovim/build/include
Compiled by runner@Mac-1658293903060.local

```

</details>

**Comment by glepnir** [comment]
@changyiyao  I want to know  what's filetype in this buffer

**Comment by johnallen3d** [comment]
@glepnir - I've been seeing this message frequently as well ("All servers ... text/documentSymbol") in various filetypes (most common `lua` and `python). I decided to dig in on what was causing it and it turns to be `null-ls`. Basically, if I have a client attached to a `null-ls` provided "server" [this message](https://github.com/glepnir/lspsaga.nvim/blob/7559a32a527c6d18b8e8ddaca64fd4ff960b001d/lua/lspsaga/symbolwinbar.lua#L163) is raised. I've managed to make it less disruptive by setting `vim.notify = require("notify")` (via `nvim-notify`) but I'd love to understand what I'm doing wrong here.

Happy to open a new issue if that's preferred.
